@here/harp-mapview/lib/MapView.ts

```diff
@@ -2133,54 +2136,74 @@ export class MapView extends EventDispatcher {
      * @param dataSource - The data source.
      */
     async addDataSource(dataSource: DataSource): Promise<void> {
-        const twinDataSource = this.getDataSourceByName(dataSource.name);
-        if (twinDataSource !== undefined) {
-            throw new Error(
-                `A DataSource with the name "${dataSource.name}" already exists in this MapView.`
-            );
-        }
+        const dsPromise = new Promise<void>(async resolve => {
```

Comment:
which addDataSource promises do you want to catch here?
Are we not just interessted in DataSources added internally without knowledge of the user? Then you could add the promise to the array where the actual addDataSource call happens, not internally here.

Reply:
The problem is that it happens in the MapViewEnvironment, so I would somehow need to expose it externally.

Reply:
Hmm, then maybe we should recheck how we create the BackgroundDataSource, is it actually async, I think its neither connecting to anything nor loading not needing a theme... so it should be sync.. maybe we can solve how we handle the BackgroundDataSource...

Reply:
Yes, if we can sync its creation, and avoid the `loadPromise`, that would be best.